toffy/watcher_callbacks.py

```diff
@@ -155,3 +196,57 @@ def fov_callback(run_folder: str, point_name: str):
                 )
 
     return fov_callback
+
+
+def build_callbacks(run_callbacks: Iterable[str],
+                    fov_callbacks: Iterable[str] = ('extract_tiffs',), **kwargs):
+    """Deduces and assembles all run & FoV callbacks for the watcher function
+
+    Args:
+        run_callbacks (Iterable[str]):
+            List of run callback names.  These will deduce the prerequisite fov callbacks
+        fov_callbacks (Iterable[str]):
+            List of fov callbacks to be run, regardless of prerequisite status
+        **kwargs (Dict[str, Any]):
+            Arguments to pass to `RunCallbacks` and `FovCallbacks` member functions
+
+    Raises:
+        ValueError:
+            Raised on non-existant member function or missing required kwarg
+
+    Returns:
+        Callable[[None,], None], Callable[[str, str], None]:
+            Assembled run callback and fov callback
+    """
+
+    methods = [attr for attr in dir(RunCallbacks) if attr[0] != '_']
+
+    fov_callbacks = set(fov_callbacks)
+
+    misc_utils.verify_in_list(requested_callbacks=run_callbacks, valid_callbacks=methods)
+    for run_cb in run_callbacks:
+        argnames = inspect.getfullargspec(getattr(RunCallbacks, run_cb))[0]
+        argnames = [argname for argname in argnames if argname != 'self']
+
+        misc_utils.verify_in_list(
+            required_arguments=argnames,
+            passed_arguments=list(kwargs.keys())
+        )
+
+        fov_callbacks = fov_callbacks.union(RUN_PREREQUISITES.get(run_cb, set()))
+
+    fov_callback = build_fov_callback(*list(fov_callbacks), **kwargs)
+
+    def run_callback():
```

Comment:
Do you think `run_callback()` should have it's own builder function such as how `fov_callback` has `build_fov_callback`?

Reply:
it could in theory.  this way seemed easier, since otherwise I'd pass in the `fov_callbacks` argument, update it from within said `build_run_callback()` function, and return it along with the constructed run callback.  

I could alternatively keep the fov_callback prerequisite checking here, but that would mean redundantly looping over the run_callbacks + verifying them again.

Regardless, I'll go with the majority opinion

Reply:
Oh I see why you did it this way. It makes sense!